app/controllers/solidus_square/callback_actions_controller.rb

```diff
@@ -11,21 +11,31 @@ def square_checkout
         access_token: SolidusSquare.config.square_access_token,
         environment: SolidusSquare.config.square_environment,
         location_id: SolidusSquare.config.square_location_id
-      ).checkout(order, redirect_url)[:checkout_page_url]
+      ).checkout(current_order, redirect_url)[:checkout_page_url]
 
       respond_to do |format|
         format.html { redirect_to checkout_page_url }
       end
     end
 
+    def complete_checkout
+      @current_order = ::Spree::Order.create(user_id: current_order.user_id)
+      cookies.signed[:guest_token] = current_order.guest_token
+      redirect_to preferred_redirect_url
+    end
+
     private
 
     def order
       @order ||= current_order
     end
 
     def redirect_url
-      ::SolidusSquare::PaymentMethod.active.first&.preferred_redirect_url
+      spree_url << "complete_checkout"
```

Comment:
What is this? Why you are not using the UrlHelper here? I know that the route can be disabled, but I think you should add it to the route table only during the test.

Reply:
https://github.com/nebulab/solidus_square/pull/37/commits/917128edce435cd2fb5c75976678be8f8a5881cf